WebPhapp/backend/server.js

```diff
@@ -215,54 +109,159 @@ app.post('/api/v1/prescriptions/add',(req,res) => {
+        if (prescription.patientID === patientID) toSend.push(prescription);
     });
 
-    // '==' catches both null and undefined
-    if (prescription == null) {
-        console.log('/api/v1/prescriptions/single: No ID match');
-        res.json({});
+    // if no prescriptions for a patient ID, return early
+    var msg = 'Sent ' + toSend.length.toString() +
+                ' prescription(s) for patient ID ' + patientID.toString();
+    if (toSend.length === 0) {
+        console.log(msg);
+        res.json([]);
         return;
     }
 
     // if no connection string (Travis testing), fill drugName with dummy info
     if (!conn.MySQL) {
-        prescription.drugName = "drugName";
-        res.json(prescription);
+        for (var i = 0; i < toSend.length; i++){
+            toSend[i].drugName = "drugName";
+        }
+        res.json(toSend);
         return;
     }
 
-    // Look up the drug name given the ID in MySQL
-    getDrugNamesFromIDs([prescription.drugID])
+    // Look up the drug names given the list of drugIDs in MySQL
+    var drugIDs = toSend.map((prescription) => {
+        return prescription.drugID;
+    })
+
+    mysql.getDrugNamesFromIDs(drugIDs, connection)
     .then((answer) => {
-        if (answer.rows.length === 0) {
-            prescription.drugName = '';
-        } else {
-            prescription.drugName = answer.rows[0].NAME;
+        for (var i = 0; i < toSend.length; i++){
+            var drug = answer.rows.filter((row) => {
+                return (row.ID === toSend[i].drugID);
+            });
+
+            // Could be undefined on return
+            if(drug.length !== 0)
+              toSend[i].drugName = drug[0].NAME;
         }
 
-        console.log("/api/v1/prescriptions/single: Sent prescription with ID " + prescriptionID);
-        res.json(prescription);
+        console.log(msg);
+        res.json(toSend);
     })
     .catch((error) => {
-        console.log("/api/v1/prescriptions/single: error: ", error);
+        console.log("/api/v1/prescriptions: error: ", error);
         res.json({});
     });
 });
 
+/*
+An api endpoint that returns a single prescription given a prescription ID
+Examples:
+    Directly in terminal:
+        >>> curl "http://localhost:5000/api/v1/prescriptions/single/1"
+    To be used in Axois call:
+        .get("api/v1/prescriptions/single/1")
+Returns:
+    A prescription object each with fields: [
+        prescriptionID,
+        patientID,
+        drugID,
+        fillDates,
+        writtenDate,
+        quantity,
+        daysFor,
+        refillsLeft,
+        prescriberID,
+        dispenserID,
+        cancelDate,
+        drugName
+    ]
+Warning:
+    no validation exists for blockchain index yet. See Issue #32 on GitHub.
+*/
+app.get('/api/v1/prescriptions/single/:prescriptionID', (req,res) => {
+    var prescriptionID = parseInt(req.params.prescriptionID);
+    var handlePrescriptionCallback = function(prescription) {
```

Comment:
Should `handlePrescriptionCallback` be its own utility function? This seems like a  useful thing to have.